specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,313 @@
+    if (dropHelper)
+    {
+        dropHelper->DragEnter(GetHWND(), dataObject, &point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->DragEnter(
+        dataObject, keyState, point, effect);
+}
+
+HRESULT DropTarget::DragOver(DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we moved over it so it can update the drag image
+    // and get the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->DragOver(&point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    // This returns whether the resultant point is over the WebView visual.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->DragOver(
+        keyState, point, effect);
+}
+
+HRESULT DropTarget::DragLeave()
+{
+    // Tell the helper that we moved out of it so it can update the drag image.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->DragLeave();
+    }
+
+    return m_webViewCompositionController2->DragLeave();
+}
+
+HRESULT DropTarget::Drop(IDataObject* dataObject,
+    DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we dropped onto it so it can update the drag image
+    // and get the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->Drop(dataObject, &point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    // This returns whether the resultant point is over the WebView visual.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->Drop(
+        dataObject, keyState, point, effect);
+}
+```
+
+## WinRT
+```c#
+// WinRT Sample
+private void WebView_DragEnter(object sender, DragEventArgs e)
+{
+  uint keyboardState = 
+    ConvertDragDropModifiersToWin32KeyboardState(e.Modifiers);
+  Point pointerPosition = CoreWindow.GetForCurrentThread().PointerPosition;
+  DataPackageOperation operation = 
+    (DataPackageOperation)webView2CompositionController.DragEnter(
+      e.Data, keyboardState, pointerPosition);
+  e.AcceptedOperation = operation;
+}
+
+private void WebView_DragOver(object sender, DragEventArgs e)
+{
+  uint keyboardState =
+    ConvertDragDropModifiersToWin32KeyboardState(e.Modifiers);
+  Point pointerPosition = CoreWindow.GetForCurrentThread().PointerPosition;
+  DataPackageOperation operation = 
+    (DataPackageOperation)webView2CompositionController.DragOver(
```

Comment:
Don't we need to pass the `e.Data` too? Same for `DragLeave`. So that the `webView2CompositionController` can clean up any temporary state it had set on the data object. (E.g., removing the drop description.)